scheduler/workflow.go

```diff
@@ -431,7 +452,12 @@ func submitPublishJob(pj job, t *task, wg *sync.WaitGroup, pu *publishNode) {
 	// Decrement the waitgroup
 	defer wg.Done()
 	// Create a new process job
-	j := newPublishJob(pj, pu.Name(), pu.Version(), pu.InboundContentType, pu.config.Table(), t.metricsManager, t.id)
+	var j job
+	if pu.Hostname != "" {
+		j = newPublishJob(pj, pu.Name(), pu.Version(), pu.InboundContentType, pu.config.Table(), t.RemoteManagers.Get(pu.Hostname), t.id)
```

Comment:
>So in my opinion, we should either have everything communicate over RPC (local and remote) or keep everything local communicating over in-process, but we shouldn't mix and match.

I feel that we have the opportunity to significantly optimize here and its low hanging fruit.  The REST package could/should take an instance of the shim, just like the scheduler does today, so if there is no need to talk remote a remote client will not be generated.  This would also give us the path provide a switch that would enforce local only communication (no gPRC).